src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/SentenceValidationService.kt

```diff
@@ -0,0 +1,52 @@
+package uk.gov.justice.digital.hmpps.calculatereleasedatesapi.service
+
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.exceptions.UnsupportedSentenceException
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.model.SentenceAndOffences
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.model.SupportedSentence
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.util.isAfterOrEqualTo
+
+@Service
+class SentenceValidationService {
+  fun validateSupportedSentences(sentenceAndOffences: List<SentenceAndOffences>) {
+    val unsupportedSentences = sentenceAndOffences.filter { !canSentenceBeCalculated(it) }
+    if (unsupportedSentences.isNotEmpty()) {
+      throw UnsupportedSentenceException(
+        "Calculation engine cannot calculate these sentences ${unsupportedSentences.map(SentenceAndOffences::sentenceCalculationType).joinToString()}",
+        unsupportedSentences
+      )
+    }
+  }
+
+  private fun canSentenceBeCalculated(sentenceAndOffence: SentenceAndOffences): Boolean {
+    return supportedSentences.any { it.sentenceType == sentenceAndOffence.sentenceCalculationType }
+  }
+
+  companion object {
+    private val supportedSentences: List<SupportedSentence> = listOf(
+      SupportedSentence("ADIMP", offenceDateCondition = { it.isAfterOrEqualTo(ImportantDates.CJA_DATE) }),
```

Comment:
See my comment about using an enum for SupportedSentence and generating a list from the enum.. 
A bit tidier codewise if instead of writing 
ImportantDates.CJA_DATE
you just write
`CJA_DATE`
and import it

Reply:
Done thanks.